python/vyos/ifconfig/ethernet.py

```diff
@@ -310,22 +310,35 @@ def set_rps(self, state):
         rps_cpus = 0
         queues = len(glob(f'/sys/class/net/{self.ifname}/queues/rx-*'))
         if state:
+            cpu_count = os.cpu_count()
```

Comment:
`os.cpu_count()` still returns numbers un-adjusted for SMT like Intel Hyper-Threading. We have a function that returns physical CPUs only in ([vyos.utils.cpu](https://github.com/vyos/vyos-1x/blob/current/python/vyos/utils/cpu.py#L84)).

I'm not completely sure if the distinction can be truly important here, but maybe use that.

Reply:
As we do not know which cores are real cores it is useless
In this case we interested in total cores and calculate a bit mask for CPU’s 
And it is not new, we used this before 

this way it could be the worse thing if you have 4 CPU and 2 threads 
For example real cores could be 0,2,4,6 and logical cores 1,3,5,7. What will be if you calculate only 4 cores (which is mix in this case of real/logical cores)?
In this case it will process by 2 real and 2 logical cores instead of all cores.